pkg/runtime/runtime_test.go

```diff
@@ -5469,30 +5469,25 @@ func TestGracefulShutdownPubSub(t *testing.T) {
 	mockAppChannel := new(channelt.MockAppChannel)
 	rt.appChannel = mockAppChannel
 	mockAppChannel.On("InvokeMethod", mock.MatchedBy(matchContextInterface), matchDaprRequestMethod("dapr/subscribe")).Return(fakeResp, nil)
-
 	require.NoError(t, rt.initPubSub(cPubSub))
 	mockPubSub.AssertCalled(t, "Init", mock.Anything)
 	rt.startSubscriptions()
 	mockPubSub.AssertCalled(t, "Subscribe", mock.AnythingOfType("pubsub.SubscribeRequest"), mock.AnythingOfType("pubsub.Handler"))
-	assert.NotNil(t, rt.pubsubCtx)
-	assert.NotNil(t, rt.topicCtxCancels)
-	assert.NotNil(t, rt.topicRoutes)
+	assert.Nil(t, rt.pubsubCtx.Err())
 	rt.running.Store(true)
 	go sendSigterm(rt)
 	select {
 	case <-rt.pubsubCtx.Done():
-		assert.Nil(t, rt.pubsubCtx)
-		assert.Nil(t, rt.topicCtxCancels)
-		assert.Nil(t, rt.topicRoutes)
+		assert.NotNil(t, rt.pubsubCtx.Err()) // check that the context is not active
```

Comment:
```suggestion
		assert.NoError(t, rt.pubsubCtx.Err()) // check that the context is not active
```

Reply:
@ItalyPaleAle - I am actually expecting an Error in this case as the context would have been cancelled during stopSubscription - here Err() method provides the cancellation reason if the context is not active and hence the check for notNil.
I have updated this to assert for the specific error - 
```code
	case <-rt.pubsubCtx.Done():
		assert.Error(t, rt.pubsubCtx.Err(), context.Canceled)
```